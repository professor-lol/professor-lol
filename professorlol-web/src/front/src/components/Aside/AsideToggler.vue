<template>
  <button
    :display="display"
    :mobile="mobile"
    :class="classList"
    type="button"
    @click="asideToggle">
    <span class="navbar-toggler-icon" />
  </button>
</template>

<script>
import { asideMenuCssClasses, validBreakpoints, checkBreakpoint } from '../../shared/classes'
import toggleClasses from '../../shared/toggle-classes'

export default {
  name: 'AsideToggler',
  props: {
    defaultOpen: {
      type: Boolean,
      default: false
    },
    display: {
      type: String,
      default: 'lg'
    },
    mobile: {
      type: Boolean,
      default: false
    },
  },
  computed: {
    classList () {
      return [
        'navbar-toggler'
      ]
    }
  },
  mounted: function() {
    this.toggle(this.defaultOpen)
  },
  methods: {
    toggle (force) {
      const [display, mobile] = [this.display, this.mobile]
      let cssClass = asideMenuCssClasses[0]
      if (!mobile && display && checkBreakpoint(display, validBreakpoints)) {
        cssClass = `aside-menu-${display}-show`
      }
      toggleClasses(cssClass, asideMenuCssClasses, force)
    },
    asideToggle (e) {
      e.preventDefault()
      this.toggle()
    }
  }
}
</script>
