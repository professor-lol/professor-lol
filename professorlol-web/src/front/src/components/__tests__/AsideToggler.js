import { mount } from "@vue/test-utils";
import AsideToggler from "../Aside/AsideToggler";

describe("AsideToggler.vue", () => {
  // Inspect the raw component options
  it('has toggle method', () => {
    expect(typeof AsideToggler.methods.toggle).toBe('function')
  })
  it('has asideToggle method', () => {
    expect(typeof AsideToggler.methods.asideToggle).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(AsideToggler)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.is('button')).toBe(true)
    expect(wrapper.classes()).toContain('navbar-toggler')
  })
});
