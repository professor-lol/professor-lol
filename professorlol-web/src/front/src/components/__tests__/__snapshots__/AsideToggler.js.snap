// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideToggler.vue renders correctly 1`] = `
<button
  class="navbar-toggler"
  display="lg"
  type="button"
>
  <span
    class="navbar-toggler-icon"
  />
</button>
`;
